### ISSUE TITLE
Homebridge Characteristic Warnings & Homekit not working

### ISSUE THREAD

**Issue by suburbanidiot** [KyleBoyer/homebridge-blaq#4]
**Describe The Bug:**
Minor issues - posting just in case someone else sees the same issue.  I'm getting characteristic warnings from homebridge & homekit status/connectivity issues after upgrading to v0.2.32

**To Reproduce:**
Issue 1: Characteristic Warnings (https://github.com/homebridge/homebridge/wiki/Characteristic-Warnings) only seen at homebridge reboot.

Issue 2: HomeKit status/connectivity issues: Only solution I found was to remove the bridge from homekit & uninstall/reinstall the plugin.  This solves the status & connectivity issues in homekit (built in blaq admin page remained working).  Note: No log errors for homekit connectivity issues.  HomeKit just stopped working.

**Logs:**

```
[8/15/2024, 9:18:55 AM] [homebridge-blaq] This plugin generated a warning from the characteristic 'Position State': Characteristic not in required or optional characteristic section for service GarageDoorOpener. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[8/15/2024, 9:18:55 AM] [homebridge-blaq] This plugin generated a warning from the characteristic 'Hold Position': Characteristic not in required or optional characteristic section for service GarageDoorOpener. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[8/15/2024, 9:18:55 AM] [homebridge-blaq] This plugin generated a warning from the characteristic 'Current Position': Characteristic not in required or optional characteristic section for service GarageDoorOpener. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
[8/15/2024, 9:18:55 AM] [homebridge-blaq] This plugin generated a warning from the characteristic 'Target Position': Characteristic not in required or optional characteristic section for service GarageDoorOpener. Adding anyway.. See https://homebridge.io/w/JtMGR for more info.
```

**Plugin Config:**
Manually assigned IP to 2 blaq devices

```json snippet
{
            "name": "Homebridge BlaQ",
            "devices": [
                {
                    "displayName": "Left Garage Door",
                    "host": "192.168.x.xxx",
                    "port": 80
                },
                {
                    "displayName": "Right Garage Door",
                    "host": "192.168.x.xxx",
                    "port": 80
                }
            ],
            "enableNativeAPIHeartbeat": true,
            "garageDoorType": "garage",
            "enableSeparateObstructionSensor": true,
            "enableMotionSensor": true,
            "enablePreCloseWarning": true,
            "enableLockRemotes": true,
            "enableLight": true,
            "enableLearnMode": true,
            "platform": "BlaQHomebridgePlugin",
            "_bridge": {
                "username": "MAC ADDRESS I DON'T RECOGNIZE",
                "port": 48975
            }
        }
```

**Environment:**

* **Plugin Version**: v0.2.32
* **Homebridge Version**: 1.8.4
* **Node.js Version**: v20.16.0
* **NPM Version**: 10.8.1
* **Operating System**: Debian GNU/Linux Bookworm (12)

**Comment by suburbanidiot** [comment]
Please note: Everything is working after uninstall / reinstall.  "Characteristic warnings" remain, but not seeing any degraded homebridge performance.  Not sure why the upgrade didn't work but the reinstall did (no log errors before or after).  Hope this helps.

**Comment by KyleBoyer** [comment]
Those characteristic warnings exist because we do publish some extra characteristics on the garage door, for positional control. These characteristics might not be recognized by HomeKit or HomeBridge, but third party applications can take advantage of these - such as "Controller for HomeKit", or "Eve".

![image](https://github.com/user-attachments/assets/0750fe24-1ed3-4fef-a460-485dc10c3153)

**Comment by KyleBoyer** [comment]
For Issue #2, did other HomeBridge plugins stop working in HomeKit? Is this a repeating thing? Is it possible you disconnected from WiFi and do not have a Home Hub to access your HomeKit when off the local network?

**Comment by suburbanidiot** [comment]
> Those characteristic warnings exist because we do publish some extra characteristics on the garage door, for positional control. These characteristics might not be recognized by HomeKit or HomeBridge, but third party applications can take advantage of these - such as "Controller for HomeKit", or "Eve".
> 
> ![image](https://private-user-images.githubusercontent.com/1487324/358277986-0750fe24-1ed3-4fef-a460-485dc10c3153.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjM3MzU3OTQsIm5iZiI6MTcyMzczNTQ5NCwicGF0aCI6Ii8xNDg3MzI0LzM1ODI3Nzk4Ni0wNzUwZmUyNC0xZWQzLTRmZWYtYTQ2MC00ODVkYzEwYzMxNTMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MDgxNSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDA4MTVUMTUyNDU0WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9ZjVlNjM3MzdhZTgyYWY5M2IxYTU2MmQ2NjA5YjUxOGY3ZTNlNGY2OGM3ZGViNzNlNzg0ODE0MzczMWIzZjYyZiZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QmYWN0b3JfaWQ9MCZrZXlfaWQ9MCZyZXBvX2lkPTAifQ.eoIlGn7kdxlqxIDa0vcVeaWLh_TrJ2c7NUr0Uc5HM_Y)

Makes sense.  I just didn’t see any other logged errors, so I figured I would report.

**Comment by suburbanidiot** [comment]
> For Issue #2, did other HomeBridge plugins stop working in HomeKit? Is this a repeating thing? Is it possible you disconnected from WiFi and do not have a Home Hub to access your HomeKit when off the local network?

No other issues from HomeKit.  Other homebridge plugins continued to function and native devices functioned correctly as well.

**Comment by suburbanidiot** [comment]
I'm going to close this bug unless you think there is something here.  My guess it was a random upgrade error.  Only thing I have seen in the log since I got it all working again is as follows.
...
[8/15/2024, 3:31:27 PM] [Homebridge BlaQ] EventSource got error Event { type: 'error', message: 'read ECONNRESET' }
[8/15/2024, 3:31:27 PM] [Homebridge BlaQ] Reinitializing EventSource...
[8/15/2024, 3:32:47 PM] [Homebridge BlaQ] EventSource got error Event { type: 'error', message: 'read ECONNRESET' }
[8/15/2024, 3:32:47 PM] [Homebridge BlaQ] Reinitializing EventSource...
[8/15/2024, 3:34:13 PM] [Homebridge BlaQ] EventSource got error Event { type: 'error', message: 'read ETIMEDOUT' }
[8/15/2024, 3:34:13 PM] [Homebridge BlaQ] Reinitializing EventSource...
...
